common/pipeline_storage/src/test/scala/weco/pipeline_storage/EitherIndexerTest.scala

```diff
@@ -32,56 +34,77 @@ class EitherIndexerTest
+  )(
+    implicit leftEncoder: Encoder[L],
+    rightEncoder: Encoder[R]
+  ): Unit = {
+    withLocalInitialImagesIndex {
+      imageIndex =>
+        withLocalMergedWorksIndex {
+          workIndex =>
+            val indexer = new EitherIndexer(
+              leftIndexer = new ElasticIndexer[L](
+                client = elasticClient,
+                index = workIndex
+              ),
+              rightIndexer = new ElasticIndexer[R](
+                client = elasticClient,
+                index = imageIndex
+              )
+            )
+            testWith((indexer, workIndex, imageIndex))
+        }
+    }
 
-        whenReady(indexer(worksAndImages)) { result =>
-          result shouldBe a[Right[_, _]]
+  }
 
-          assertElasticsearchEventuallyHas(workIndex, works: _*)
-          assertElasticsearchEventuallyHas(imageIndex, images: _*)
+  it("indexes a list of either works or images") {
+    withEitherIndexer[Work[Merged], Image[Initial]] {
+      case (indexer, workIndex, imageIndex) =>
+        whenReady(indexer(worksAndImages)) {
+          result =>
+            result shouldBe a[Right[_, _]]
+            assertElasticsearchEventuallyHas(workIndex, works: _*)
+            assertElasticsearchEventuallyHas(imageIndex, images: _*)
         }
-      }
     }
   }
 
   it("indexes a list of only works") {
-    withLocalInitialImagesIndex { imageIndex =>
-      withLocalMergedWorksIndex { workIndex =>
-        val indexer = new EitherIndexer(
-          leftIndexer = new ElasticIndexer[Work[Merged]](
-            client = elasticClient,
-            index = workIndex,
-            config = WorksIndexConfig.merged
-          ),
-          rightIndexer = new ElasticIndexer[Image[Initial]](
-            client = elasticClient,
-            index = imageIndex,
-            config = ImagesIndexConfig.initial
-          )
-        )
+    withEitherIndexer[Work[Merged], Image[Initial]] {
+      case (indexer, workIndex, _) =>
+        whenReady(indexer(works.map(Left(_)))) {
+          result =>
+            result shouldBe a[Right[_, _]]
+            assertElasticsearchEventuallyHas(workIndex, works: _*)
+        }
+    }
+  }
+
+  it("indexes a list of only images") {
+    withEitherIndexer[Work[Merged], Image[Initial]] {
+      case (indexer, _, imageIndex) =>
+        whenReady(indexer(images.map(Right(_)))) {
+          result =>
+            result shouldBe a[Right[_, _]]
+            assertElasticsearchEventuallyHas(imageIndex, images: _*)
+        }
+    }
+  }
 
-        whenReady(indexer(works.map(Left(_)))) { result =>
-          result shouldBe a[Right[_, _]]
-          assertElasticsearchEventuallyHas(workIndex, works: _*)
+  it("returns successfully when there are neither works nor images") {
+    withEitherIndexer[Work[Merged], Image[Initial]] {
+      case (indexer, _, _) =>
+        whenReady(indexer(Nil)) {
+          result =>
+            result shouldBe a[Right[_, _]]
```

Comment:
I spotted a couple of notable omissions from this suite - only images and nothing, so here they are.

There is also no test for failures, but that wouldn't follow the same pattern and would add even more change to an already rather large PR.